add_nyt_bestseller_tag.py

```diff
@@ -0,0 +1,102 @@
+"""
+A bot to add a NYT bestseller tag 'nyt:{list_name_encoded}={published_date}' to each book from the list of isbns.
+The list of bestsellers to add is a json file of the following format:
+[
+    {
+        "list_name_encoded": "business-books",
+        "published_date": "2020-07-12",
+        "isbns": [
+            "9780735211292",
+            "9780399592522"
+        ]
+    },
+]
+This script would be called from the command line like so:
+$ python add_nyt_bestseller_tag.py --file=~/nyt_bstslrs.json --limit=1 --dry-run=True
+NOTE: This script checks if there is a tag on a book that starts with
+'nyt:' and if the book has such a tag the script does nothing.
+If there is no such book with requested isbn in the OL, the script makes request
+by the book isbn https://openlibrary.org/isbn/{isbn} , triggering autoimport
+"""
+
+import json
+import urllib
+
+from olclient.bots import AbstractBotJob
+
+class AddNytBestsellerJob(AbstractBotJob):
+  def need_to_add_nyt_bestseller_tag(self, work) -> bool:
+    """Returns False if the book already has a tag that starts with 'nyt:'"""
+    for subj in work['subjects']:
+      if subj.startswith(('nyt:', 'nytimes:')):
+        return False
+    return True
+
+  def add_tag(self, work, new_tag) -> None:
+    """Adds a new tag to a work"""
+    if 'subjects' not in work:
+      work['subjects'] = list(new_tag)
+    else:
+      work['subjects'].append(new_tag)
+
+  def book_exists_in_ol(self,book_isbn) -> bool:
+    """Returns True if Edition with book_isbn exists. Returns False otherwise"""
+    try:
+      found_edition = self.ol.Edition.get(isbn=book_isbn)
+      if not found_edition:
+        self.logger.info('Not found edition for ISBN'+book_isbn)
+        return False
+      self.logger.info('Found edition for ISBN'+book_isbn)
+      return True
+    except Exception as e:
+      #TODO check the exception type
+      job.logger.exception('There is no book with ISBN ' + book_isbn)
+      return False
+
+  def request_book_import_by_isbn(self,book_isbn)-> None:
+    """ Makes request to the book_isbn https://openlibrary.org/isbn/{book_isbn} """
+    url = "https://openlibrary.org/isbn/%s" % (book_isbn)
+    request = (urllib.Request(url, None, headers={"Referrer": "http://www.openlibrary.org"}))
+    conn = urllib.urlopen(request)
+    try:
+      conn.read()
+      job.logger.exception('Made request to ' + url)
+    except Exception as e:
+      self.logger.info('Failed to make request to  '+url)
+    finally:
+      conn.close()
+
+  def run(self) -> None:  # overwrites the AbstractBotJob run method
+    self.dry_run_declaration()
+
+    comment = 'Add NYT bestseller tag'
+    with open(self.args.file, 'r') as fin:
+        bestsellers_data = json.load(fin)
+        if 'error' not in bestsellers_data:
```

Comment:
What is this line intended to do? Is there a situation where the string error might be in bestseller_data?

Reply:
Yeah, that's just a code I picked from old plugin, it doesn't make sense in our impl. Removed.